src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -30,58 +61,98 @@ public Drive(SwerveDriveMap map) {
-        });
+        // Now use this in our kinematics
+        final SwerveModuleState[] moduleStates = kinematics.toSwerveModuleStates(speeds);
+
+        // Front left module state
+        io.frontLeft.desiredState = moduleStates[0];
+
+        // Front right module state
+        io.frontRight.desiredState = moduleStates[1];
+
+        // Back left module state
+        io.rearLeft.desiredState = moduleStates[2];
+
+        // Back right module state
+        io.rearRight.desiredState = moduleStates[3];
     }
 
-    public CommandBase crossWheels() {
-        return cmd("Cross Wheels").onInitialize(() -> {
-            // speed really low to work around speed == 0 which prevents the swerve moduals
-            // turning
-            SwerveModuleState crossingMotorA = new SwerveModuleState(.01, Rotation2d.fromDegrees(45));
-            SwerveModuleState crossingMotorB = new SwerveModuleState(.01, Rotation2d.fromDegrees(-45));
-            // Front left module state
-            io.frontLeft.desiredState = crossingMotorB;
-
-            // Front right module state
-            io.frontRight.desiredState = crossingMotorA;
-
-            // Back left module state
-            io.rearLeft.desiredState = crossingMotorA;
-
-            // Back right module state
-            io.rearRight.desiredState = crossingMotorB;
-        }).runsUntil(() -> Math.abs(-45 - io.frontLeft.podAngle) < 1 && Math.abs(45 - io.rearRight.podAngle) < 1 && Math
-                .abs(-45 - io.rearLeft.podAngle) < 1 && Math.abs(45 - io.frontRight.podAngle) < 1);
+    public CommandBase drive(DoubleSupplier xSpeed, DoubleSupplier ySpeed, DoubleSupplier rotation) {
+        return run(() -> move(xSpeed.getAsDouble(), ySpeed.getAsDouble(), rotation.getAsDouble()));
+    }
+
+    // Use DrivePID to drive to a target pose on the field
+    public CommandBase driveTo(Pose2d targetPose) {
+        return cmd().onExecute(() -> {
+            Transform2d fb = drivePID.calculate(pose, targetPose);
+            move(fb.getX(), fb.getY(), fb.getRotation().getDegrees());
+        }).runsUntil(() -> drivePID.isFinished(pose, targetPose, 0.005)).onEnd(() -> move(0, 0, 0));
+    }
+
+    // Drive to a pre-determined grid position
+    public CommandBase driveTo(GridPosition gridPose) {
+        return driveTo(gridPose.getPose());
+    }
+
+    private Pose2d closestPose = GridPosition.values()[0].getPose();
+
+    // Find the nearest grid position and line up with it
+    public CommandBase driveToNearest() {
+
+        return cmd().onInitialize(() -> {
+            closestPose = GridPosition.values()[0].getPose();
+            for (GridPosition position : GridPosition.values()) {
+                if (position.getPose().getTranslation().getDistance(pose.getTranslation()) < closestPose
+                        .getTranslation().getDistance(pose.getTranslation())) {
+                    closestPose = position.getPose();
+                }
+            }
+        }).onExecute(() -> {
```

Comment:
It would be really nice if you could use `driveTo` here. If you use [`ProxyCommand`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/ProxyCommand.html#%3Cinit%3E(java.util.function.Supplier)) then you can get the pose and create the command from the lambda